src/query/v2/multiframe.cpp

```diff
@@ -0,0 +1,115 @@
+// Copyright 2022 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include "query/v2/multiframe.hpp"
+
+#include "query/v2/bindings/frame.hpp"
+#include "utils/pmr/vector.hpp"
+
+namespace memgraph::query::v2 {
+
+// #NoCommit uncomment https://github.com/memgraph/memgraph/pull/676#discussion_r1035704661
+// static_assert(std::forward_iterator<ValidFramesReader::Iterator> &&
+//               std::equality_comparable<ValidFramesReader::Iterator>);
+// static_assert(std::forward_iterator<ValidFramesModifier::Iterator> &&
+//               std::equality_comparable<ValidFramesModifier::Iterator>);
+// static_assert(std::forward_iterator<ValidFramesConsumer::Iterator> &&
+//               std::equality_comparable<ValidFramesConsumer::Iterator>);
+// static_assert(std::forward_iterator<InvalidFramesPopulator::Iterator> &&
+//               std::equality_comparable<InvalidFramesPopulator::Iterator>);
+
+MultiFrame::MultiFrame(int64_t size_of_frame, size_t number_of_frames, utils::MemoryResource *execution_memory)
+    : frames_(utils::pmr::vector<FrameWithValidity>(
+          number_of_frames, FrameWithValidity(size_of_frame, execution_memory), execution_memory)) {
+  MG_ASSERT(number_of_frames > 0);
+}
+
+MultiFrame::MultiFrame(const MultiFrame &other) {
+  frames_.reserve(other.frames_.size());
+  std::transform(other.frames_.begin(), other.frames_.end(), std::back_inserter(frames_),
+                 [](const auto &other_frame) { return other_frame; });
+}
```

Comment:
```suggestion
MultiFrame::MultiFrame(const MultiFrame &other) :frames_{other.frames_} {}
```

Please format to be sure it is fine.